front/deployfile_download.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ * GLPI Inventory Plugin
+ * Copyright (C) 2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on FusionInventory for GLPI
+ * Copyright (C) 2010-2021 by the FusionInventory Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI Inventory Plugin.
+ *
+ * GLPI Inventory Plugin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI Inventory Plugin is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with GLPI Inventory Plugin. If not, see <https://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+use org\bovigo\vfs\FilenameTestCase;
+
+include("../../../inc/includes.php");
+
+Session::checkRight('plugin_glpiinventory_package', READ);
+global $CFG_GLPI;
+
+$filename = null;
+if (isset($_GET['filename'])) {
+    $filename = $_GET['filename'];
+}
+
+$mimetype = null;
+if (isset($_GET['mimetype'])) {
+    $mimetype = urldecode($_GET['mimetype']);
+}
```

Comment:
As far as I understand, `PluginFusioninventoryDeployFile` entry could be retrieved from using given `sha512`. Instead of passing these information in URL, they should be get from `name` and `mimetype` properties of the corresponding `PluginGlpiinventoryDeployFile` entry.

Reply:
done

Reply:
I was suggesting the opposite way. Passing the `sha512` in URL (it is URL safe) and getting the filename and mimetype from DB (filename can contains special chars that would require to use an URL encoding method).

Reply:
We should probably just use the `id` from `glpi_plugin_glpiinventory_deployfiles` table.